contrib_bots/lib/sched.py

```diff
@@ -0,0 +1,186 @@
+                return True
+            except ValueError:
+                return False
+
+        # List needed to verify if time interval was valid.
+        weekdays = ["mondays", "tuesdays", "wednesdays", "thursdays",
+                    "fridays", "saturdays", "sundays", "days"]
+
+        weekday = ["monday", "tuesday", "wednesday", "thursday",
+                   "friday", "saturday", "sunday", "day"]
+
+        intervals = ["hour", "hours", "minute", "minutes"]
+        # Verify that @every command meets the syntax required for data indexing with weekdays.
+        if len(original_content.split()) > 3:
+            if (original_content.startswith('@every') and original_content.split()[1].isdigit() and
+                (original_content.split()[2] in weekdays or original_content.split()[2] in weekday)
+               and isTimeFormat(original_content.split()[4])):
+                data = json.load(open("events.json", "rb"))
+                if key not in data:
+                    data.append(key)
+                json.dump(data, open("events.json", "wb"))
+                coefficient = int(original_content.split()[1])
+                interval = original_content.split()[2]
+                timer = original_content.split()[4]
+                if interval in weekdays and coefficient < 2:
+                    interval = interval[:-1]
+                if coefficient > 1 and interval in weekday:
+                    interval = interval+"s"
+                task = schedule.every(coefficient)
+                task = getattr(task, interval)
+                task.at(timer).do(job, a).tag(a)
+
+        # Verify with smaller intervals
+        if len(original_content.split()) > 2:
+            if (original_content.startswith('@every') and original_content.split()[2] in intervals):
+                data = json.load(open("events.json", "rb"))
+                if key not in data:
+                    data.append(key)
+                json.dump(data, open("events.json", "wb"))
+                coefficient = int(original_content.split()[1])
+                interval = original_content.split()[2]
+                if (coefficient < 2 and interval.endswith("s")):
+                    interval = interval[:-1]
+                if (coefficient > 1 and interval.endswith("s") == False):
+                    interval = interval+"s"
+                task = schedule.every(coefficient)
+                task = getattr(task, interval)
+                task.do(job, a).tag(a)
+
+        # Cancel command will remove events from persistence list and from the current schedule.
+        if original_content.startswith('@cancel'):
+            data = json.load(open("events.json", "rb"))
+            id = original_content.split()[1]
+            if id.isnumeric():
+                for i in data:
+                    try:
+                        if id == dict(i)["content"].split()[-1]:
+                            data.remove(i)
```

Comment:
You are removing an item from the iterable you are traversing, that could cause problems.